lms/djangoapps/course_blocks/transformers/grades.py

```diff
@@ -0,0 +1,61 @@
+"""
+Grades Transformer
+"""
+from django.contrib.auth.models import AnonymousUser, User
+from django.test.client import RequestFactory
+from mock import patch
+
+from lms.djangoapps.courseware import module_render
+from xmodule import x_module
+from openedx.core.lib.block_structure.transformer import BlockStructureTransformer
+
+class GradesBlockTransformer(BlockStructureTransformer):
+    """
+    """
+    VERSION = 1
+
+    FIELDS_TO_COLLECT = [u'due', u'graded', u'has_score', u'weight']
+
+    @classmethod
+    def name(cls):
+        """
+        Unique identifier for the transformer's class;
+        same identifier used in setup.py.
+        """
+        return u'grades'
+
+    @classmethod
+    def collect(cls, block_structure):
+        """
+        Collects any information that's necessary to execute this
+        transformer's transform method.
+        """
+        block_structure.request_xblock_fields(*cls.FIELDS_TO_COLLECT)
+
+        cls.collect_max_scores(block_structure)
+
+    @staticmethod
+    def _iter_xmodules(block_structure):
+        request = RequestFactory().get('/dummy-collect-max-grades')
+        request.user = AnonymousUser()
+        request.session = {}
+        for block_locator in block_structure.topological_traversal():
+            course_id = unicode(block_locator.course_key)
+            usage_id = block_locator._to_deprecated_string()
+            module, __ = module_render.get_module_by_usage_id(request, course_id, usage_id)
+            yield module
+
+    @classmethod
+    def collect_max_scores(cls, block_structure):
+        # This is to get around accessing the DB to
+        with patch.object(x_module.XModule, 'save'):
```

Comment:
I agree that this is better than writing to the database or creating dummy users. This may be a bit off base because I'm not familiar with exactly how this explodes otherwise, but a couple of questions:

1. Does it make sense to patch deeper in the stack, say at the `DjangoXBlockUserStateClient` level? This current patch is XModule specific?
2. Is it possible to avoid patching by making `DjangoXBlockUserStateClient` do more or less the right thing in the case of anonymous users?